components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -272,39 +258,31 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
                 ctx,
             );
 
-        if (!this.disposables.disposed) {
-            for (const project of projects) {
-                this.disposables.push(this.subscriber.listenForPrebuildUpdates(project.id, handler));
-            }
-        }
-
-        // TODO(at) we need to keep the list of accessible project up to date
-    }
-
-    private async getAccessibleProjects() {
-        const userId = this.userID;
-        if (!userId) {
-            return [];
+        if (!this.disposables.disposed && this.userID) {
```

Comment:
Let's move the `this.userID` check to the top of the method.

Reply:
Addressed in [`bda0663` (#20565)](https://github.com/gitpod-io/gitpod/pull/20565/commits/bda06630679c0ec97edae87e969cf5d9ad453337)